app/views/play/level/PlayLevelView.coffee

```diff
@@ -835,9 +835,19 @@ module.exports = class PlayLevelView extends RootView
       when @level?.isType('web-dev') then windowHeight - controlBarHeight
       when tomeLocation is 'bottom' then Math.min(windowHeight - minTomeHeight - controlBarHeight, windowWidth / canvasAspectRatio)
       else Math.min(windowHeight - (if controlBarLocation is 'left' then controlBarHeight else 0), (windowWidth - minCodeWidth - minWorkspaceWidth - minToolboxWidth) / canvasAspectRatio)
-    canvasWidth = switch
-      when @level?.isType('web-dev') then windowWidth - minCodeWidth
-      else canvasHeight * canvasAspectRatio
+    if me.get('aceConfig').preferWideEditor or features?.china
```

Comment:
won't it be always true for china even after setting is changed?

Reply:
yup it is 😂